cpp/src/toppra/solver/qpOASES-wrapper.cpp

```diff
@@ -0,0 +1,127 @@
+#include <toppra/solver/qpOASES-wrapper.hpp>
+
+#include <qpOASES.hpp>
+
+namespace toppra {
+namespace solver {
+struct qpOASESWrapper::Impl {
+  qpOASES::SQProblem qp;
+
+  Impl(Eigen::Index nV, Eigen::Index nC)
+    : qp (nV, nC)
+  {
+    qpOASES::Options options;
+    options.printLevel = qpOASES::PL_NONE;
+
+    qp.setOptions( options );
+  }
+};
+
+qpOASESWrapper::qpOASESWrapper (const LinearConstraintPtrs& constraints, const GeometricPath& path,
+        const Vector& times)
+  : Solver (constraints, path, times)
+{
+  // Currently only support Canonical Linear Constraint
+  Eigen::Index nC = 2; // First constraint is x + 2 D u <= xnext_max, second is xnext_min <= x + 2D u
+  for (const Solver::LinearConstraintParams& linParam : constraintsParams_.lin)
+    nC += linParam.F[0].rows();
+
+  Eigen::Index nV (nbVars());
+  assert(nV == 2);
+  A_  = RMatrix::Zero(nC, nV);
+  lA_ = -Vector::Ones(nC);
+  hA_ = -Vector::Ones(nC);
+
+  impl_ = new Impl(nV, nC);
+}
+
+qpOASESWrapper::~qpOASESWrapper ()
+{
+  delete impl_;
+}
+
+bool qpOASESWrapper::solveStagewiseOptim(std::size_t i,
+        const Matrix& H, const Vector& g,
+        const Bound& x, const Bound& xNext,
+        Vector& solution)
+{
+  Eigen::Index N (nbStages());
+  assert (i <= N);
+
+  value_type INF = 1e16;
```

Comment:
I think INF should be a constant in toppra.hpp.

Reply:
Actually, this is not necessarily correct. This constraint is mainly to work around an issue qpOASES has when the problem is unconstraint.

Reply:
So I leave this here ? It could be made a class attribute if necessary.